### ISSUE TITLE
Clean up `output_activation_fn`

### ISSUE THREAD

**Issue by znado** [mlcommons/algorithmic-efficiency#158]
Right now we have `output_activation_fn` unimplemented in basically every workload. Despite an incorrect code comment in `spec.py` which should be rewritten, this is supposed to take the model logits and turn them into probabilities. For FastMRI this should just be the identity. We can likely delete it from all non-base classes, and in the base class just implement an if/elif/else chain on the type of loss, because the logic should be the same across all workloads that share the same type of loss.
